app/service/form_task_list_service.rb

```diff
@@ -132,13 +132,26 @@ def make_form_live_section
 
   def make_form_live_section_tasks
     [{
-      task_name: I18n.t("forms.task_list_#{create_or_edit}.make_form_live_section.make_live"),
-      path: @form.all_ready_for_live? ? make_live_path(@form.id) : "",
+      task_name: live_task_name,
+      path: live_path,
       status: @task_statuses[:make_live_status],
       active: @form.all_ready_for_live?,
     }]
   end
 
+  def live_task_name
+    return I18n.t("forms.task_list_#{create_or_edit}.make_form_live_section.make_changes_live") if @form.is_archived?
+
+    I18n.t("forms.task_list_#{create_or_edit}.make_form_live_section.make_live")
+  end
+
+  def live_path
+    return "" unless @form.all_ready_for_live?
+    return unarchive_path(@form.id) if @form.is_archived?
+
+    make_live_path(@form.id)
```

Comment:
The make_live controller which the make_live_path points to already has logic for switching between the normal make live view and a new `make_archived_draft_live` view. Rather than adding this logic here I think the simplest way to fix the issue Hannah described would be to change the content in that view to match that in the unarchive view.

The way this is implemented in the most recent commit means that:
- the `make_archived_draft_live.html.erb` view is never rendered, so it and the switching logic in the make_live_controller are redundant
- the back link and vallidation on the unarchive page won't take the user back to the task list, they'll take them back to the archived form page.

I'll leave it up to you whether `make_archived_draft_live.html.erb` should be removed or not, but if we do remove it we need to preserve the different back link/validation behaviour depending on where the user has come from.

Reply:
Right I think I've got it this time...